conf/gulp-tasks/bundle/bundleabstractfactory.js

```diff
@@ -0,0 +1,188 @@
+        sourcemap: true
+      },
+      plugins: [
+        resolve(),
+        commonjs({
+          include: './node_modules/**'
+        }),
+        babel({
+          babelrc: false,
+          exclude: 'node_modules/**',
+          presets: ['@babel/env']
+        })
+      ]
+    };
+    return rollup(rollupConfig)
+      .pipe(source('answers-modern.js'))
+      .pipe(replace('@@LIB_VERSION', this._getLibVersion()))
+      .pipe(dest('dist'));
+  }
+
+  /**
+   * A factory method for producing the legacy, IIFE-style SDK bundle.
+   *
+   * @returns {File} The legacy, IIFE-style SDK bundle.
+   */
+  _legacyBundleIIFE () {
+    const legacyBundleConfig = {
+      format: 'iife',
+      name: this._namespace,
+      sourcemap: true
+    };
+    return this._legacyBundle(legacyBundleConfig, 'answers.js');
+  }
+
+  /**
+   * A factory method for producing the legacy, UMD-style SDK bundle.
+   *
+   * @returns {File} The legacy, UMD-style SDK bundle.
+   */
+  _legacyBundleUMD () {
+    const legacyBundleConfig = {
+      format: 'umd',
+      name: this._namespace,
+      export: 'default',
+      sourcemap: true
+    };
+    return this._legacyBundle(legacyBundleConfig, 'answers-umd.js');
+  }
+
+  /**
+   * A helper method that can produce either variant of the legacy SDK bundle.
+   *
+   * @param {Object<string, ?>} outputConfig Any variant-specific configuration
+   *                                         for the legacy bundle.
+   * @param {string} fileName The filename of the created bundle.
+   * @returns {File} The legacy SDK bundle.
+   */
+  _legacyBundle (outputConfig, fileName) {
+    const rollupConfig = {
+      input: './src/answers-umd.js',
+      output: outputConfig,
+      plugins: [
+        resolve(),
+        insert.prepend(
+          fs.readFileSync('./conf/gulp-tasks/polyfill-prefix.js').toString(),
+          {
+            include: './src/answers-umd.js'
+          }),
+        commonjs({
+          include: './node_modules/**'
+        }),
+        babel({
+          runtimeHelpers: true,
+          babelrc: false,
+          exclude: 'node_modules/**',
+          presets: [
+            [
+              '@babel/preset-env',
+              {
+                loose: true,
+                modules: false
+              }
+            ]
+          ],
+          plugins: [
+            '@babel/syntax-dynamic-import',
+            ['@babel/plugin-transform-runtime', {
+              corejs: 3
+            }],
+            '@babel/plugin-transform-arrow-functions',
+            '@babel/plugin-proposal-object-rest-spread'
+          ]
+        })
+      ]
+    };
+    return rollup(rollupConfig)
+      .pipe(source(fileName))
+      .pipe(replace('@@LIB_VERSION', this._getLibVersion()))
+      .pipe(dest('dist'));
+  }
+
+  /**
+   * Attempts to compute the correct library version for an SDK bundle.
+   *
+   * @returns {string} The SDK bundle's library version.
+   */
+  _getLibVersion () {
+    try {
```

Comment:
Nit but can we assign library version to a variable so it's not recomputed every time time it's called, or maybe just have it be something that's passed into this class?